docs/general/run-a-node/maintenance-guides/handling-network-upgrades.md

```diff
@@ -1,81 +1,173 @@
+fetch and validate the blocks matching the specific consensus network version.
+Each network upgrade will have detailed instructions posted in the
+corresponding [Upgrade log]. This guide describes general principles for
+network upgrades and should serve only as a supplementary.
+
 :::caution
 
-Following this guide when there is no network upgrade will result in you losing your place on the current network.
+Stopping and upgrading your node when there is no network upgrade will result
+in losing your place on the current network.
 
 :::
 
-The following guide should be used when the network has agreed to do a software upgrade.
+There are two kinds of consensus network upgrades currently in use:
 
-## Stop the node at specific epoch
+- *Seamless upgrade* requires that you update your Oasis Core without wiping the
+  consensus state or changing the genesis document.
+- *Dump & restore upgrade* requires that you wipe the state and run a new
+  version of the Oasis Core from a fresh genesis document.
 
-Before an upgrade we will update the [Upgrade Log](../upgrade-log.md) to specify the epoch height at which the upgrade will take place. Additionally, an upgrade descriptor will be provided which can be used to instruct the node to shutdown and export state at the start of the upgrade epoch.
+## Stop the node
 
-To submit the provided upgrade descriptor use the following command:
+The upgrade proposal transaction includes the *upgrade height* field. This is
+the epoch at which, once reached, your node will automatically stop, writing
+the following in the log
 
-```bash
-oasis-node control upgrade-binary \
- -a unix:/serverdir/node/internal.sock \
- <PATH-TO-UPGRADE-DESCRIPTOR.json>
+```text
+{"caller":"mux.go:426","level":"debug","module":"abci-mux","msg":"dispatching halt hooks for upgrade","ts":"2022-05-06T13:11:41.721994647Z"}
 ```
 
-To verify that the upgrade descriptor has been received, grep your node's logs for "received upgrade descriptor" message, e.g.:
-
-> {"caller":"upgrade.go:60","epoch":5102,"level":"info","module":"upgrade","msg":"received upgrade descriptor, scheduling shutdown","name":"testnet-upgrade-2021-03-24","ts":"2021-03-24T09:56:52.944552808Z"}
+and on the error output:
 
-If for any reason you would need to cancel a scheduled pending upgrade, use the `cancel-upgrade` command.
+```text
+panic: upgrade: reached upgrade epoch
+```
 
-Node reaching the upgrade epoch will automatically export network state to a genesis file under the following path:`exports/genesis-<CHAIN-ID>-at-<DUMP-BLOCK-HEIGHT>.json`.
+Then, the `oasis-node genesis dump` command will automatically execute and
+the consensus state will be written into
+`genesis-<CHAIN_ID>-at-<LATEST_HEIGHT>.json` file in the working directory,
+where `CHAIN_ID` is the chain ID of the network and `LATEST_HEIGHT` is the
+round of the last block before the upgrade epoch. This command, depending on the
```

Comment:
```suggestion
height of the consensus block at the upgrade.
```


To avoid potential confusion, don't use the term "round" when describing consensus block heights.
The term round is usually and mostly used to refer to runtime "heights".